content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,131 @@
+
+- **Streamline Hardware Integration**: Simplify code and reduce complexity.
+- **Enhance Development Speed**: Quickly set up and start working on your projects.
+- **Access Standardized APIs**: Ensure consistency across your projects.
+
+## Getting Started with ESP-BSP
+
+The following example covers steps for building application for ESP32-S3-BOX-3 which is supported by ESP-BSP.
+
+### Hardware Setup
+
+Ensure you have the following hardware:
+
+- [ESP32-S3-BOX-3](https://github.com/espressif/esp-box) development board.
+- USB-C Cable for power and programming.
+
+### Prerequisites
+
+Before you begin, make sure you have the following:
+
+- [ESP-IDF v5.3](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/index.html#installation): The official development framework for the ESP32, properly installed and sourced in your shell.
+
+### Creating Project from Example
+
+Let’s create a simple project using the `display_audio_photo` example, which is available for the ESP32-S3-BOX-3. This example showcases how to use the display, touch, and audio features.
+
+1. **Initialize a New Project**:
+
+   Use the `idf.py` tool to create a new project from the example:
+
+   ```bash
+   idf.py create-project-from-example "espressif/esp-box-3^1.2.0:display_audio_photo"
+   cd display_audio_photo
+   ```
+
+2. **Set the Target**:
+
+   Ensure the correct target is set for your project:
+
+   ```
+   idf.py set-target esp32s3
+   ```
+
+   Note: For users of ESP-IDF 5.3 or newer, it is necessary to add the following dependency with the corrected I2C driver (error message at runtime: `CONFLICT! driver_ng`):
+
+   ```
+   idf.py add-dependency "espressif/esp_codec_dev==1.1.0"
+   ```
+
+3. **Build and Flash the Project**:
+
+   Compile and flash your application to the ESP32-S3-BOX-3:
+
+   ```
+   idf.py build flash monitor
+   ```
+
+   Note: Use `Ctrl+]` to quit the monitor application.
+
+
+### Exploring the Example
+
+Once the application is running, you’ll see the following features in action:
+
+- **Display**: Shows images, text files, and more.
+- **Touch**: Interacts with the display.
+- **Audio**: Plays sound files.
+
+
+### Exploring the Component
+
+The ESP-BSP is included via `main/idf_component.yml` file. Simply by switching to another BSP, you can adapt the project to different hardware. Here’s how you can specify dependencies for different boards:
+
+Example for ESP32-S3-BOX:
+
+```
+## IDF Component Manager Manifest File
+dependencies:
+  espressif/esp-box: "^3.1.0"
+  esp_codec_dev:
+    public: true
+    version: "==1.1.0"
+  ## Required IDF version
+  idf:
+    version: ">=5.0.0"
+```
+
+Example for M5Stack-CoreS3:
+
+```
+## IDF Component Manager Manifest File
+dependencies:
+  espressif/m5stack_core_s3: "^1.1.0"
+  esp_codec_dev:
+    public: true
+    version: "==1.1.0"
+  ## Required IDF version
+  idf:
+    version: ">=5.0.0"
+```
+
```

Comment:
Maybe adding the code should help visualize the example if the code it's not that big.

Reply:
Which code do you mean? The app, it's stored as part of the example at BSP site.

Reply:
You are using the `display_audio_photo` example so that we can add some code, especially the BSP initialization.

```c
    /* Initialize and mount SPIFFS */
    bsp_spiffs_mount();

    /* Initialize I2C (for touch and audio) */
    bsp_i2c_init();

    /* Initialize display and LVGL */
    bsp_display_start();

    /* Set default display brightness */
    bsp_display_brightness_set(APP_DISP_DEFAULT_BRIGHTNESS);

    /* Add and show LVGL objects on display */
    app_disp_lvgl_show();

    /* Initialize SPI flash file system and show list of files on display */
    app_disp_fs_init();

    /* Initialize audio */
    app_audio_init();

    ESP_LOGI(TAG, "Example initialization done.");
```

Reply:
That's good idea. I'll add an paragraph

Reply:
The code is in place. Good idea, makes it easier to understand the concept of BSP API.